src/EditorFeatures/CSharpTest/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsRefactoringTests.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.UseRecursivePatterns;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings.UseRecursivePatterns
+{
+    using VerifyCS = CSharpCodeRefactoringVerifier<UseRecursivePatternsCodeRefactoringProvider>;
+
+    [Trait(Traits.Feature, Traits.Features.CodeActionsUseRecursivePatterns)]
+    public class UseRecursivePatternsRefactoringTests
+    {
+        private static Task VerifyAsync(string initialMarkup, string expectedMarkup)
+        {
+            return new VerifyCS.Test
+            {
+                LanguageVersion = LanguageVersion.CSharp9,
+                TestCode = initialMarkup,
+                FixedCode = expectedMarkup,
+                CodeActionValidationMode = CodeActionValidationMode.None,
```

Comment:
:memo: This is because the compiler see a type pattern as a constant, so when we produce a type it fails.

Reply:
@alrz So this is about that you produce a different syntax tree than what the compiler produces (but both have the same text representation)?

Reply:
Yes. type and constant trees are almost the same, it's an issue here because in patterns both are accepted in the same place.